source/javascripts/lib/wirsing/filters/switch.coffee

```diff
@@ -0,0 +1,17 @@
+@wirsing ?= {}
+wirsing.filter ?= {}
+
+wirsing.filter.switch =
+  on: (items, key) ->
+    switchObject = {}
+
+    switches = ->
+      if Object.keys(switchObject).length == 0
+        switchObject[item[key]] ?= true for item in items
+      switchObject
+
+    names: -> Object.keys switches()
```

Comment:
Please don't just start returning an object in the middle of a function, it took me 5 mins to figure it out what was going on here ;-) the code above makes it very clear that we are returning an API from the `on` method.